### ISSUE TITLE
ResolveReferencePaths on CompileLibrary instances for a deps file created for a netstandard2.0 class library throws

### ISSUE THREAD

**Issue by pranavkm** [dotnet/core-setup#3726]
Mvc attempts to load `DependencyContext` for all *available* assemblies and attempts to resolve reference paths for available CompileLibraries. Relevant code: https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/AssemblyPart.cs#L52-L56. The deps file for `netstandard2.0` targeting class libraries contain a slew of compile libraries all marked as `referenceassembly` e.g.

```
"Microsoft.Win32.Primitives/4.0.3.0": {
      "type": "referenceassembly",
      "serviceable": false,
      "sha512": ""
    },
```

Attempting to call `ResolveReferencePaths` from a CoreCLR application will fail, since the reference does not exist. Is this something we can fix or harden against?

**Comment by steveharter** [comment]
cc @eerhardt for triage

**Comment by eerhardt** [comment]
>  since the reference does not exist

What do you mean it doesn't exist?  Can you show a stand alone repro of the issue?

**Comment by eerhardt** [comment]
Is it expected that there are Razor pages in the netstandard2.0 library?  And if so, do they need to be compiled using the same ref assemblies as the rest of the code in the library?  If `yes` to the above questions, then those ref assemblies are going to need to exist at runtime somewhere, so they can be found and used for compilation.

**Comment by pranavkm** [comment]
Here's the repro app: https://github.com/pranavkm/repro-3726. Copying the deps file is a bit contrived, but that's what Mvc's test targets (that we ship to NuGet) does - https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Testing/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Testing.targets#L18-L24. The issue is that none of these references show up in the app's deps file closure because they get replaced by the Sdk with references to the package. Maybe the issue here is that we shouldn't be loading the ClassLibrary's DependencyContext in isolation, but rather read a context that's merged with the app \ runtime.

**Comment by ericstj** [comment]
> Maybe the issue here is that we shouldn't be loading the ClassLibrary's DependencyContext in isolation, but rather read a context that's merged with the app \ runtime.

That's right.  It's not like the build reuses the references of the class library in the application.  That's the whole principle behind portable (and netstandard) libraries: they provide a view of references that differs from that which is available at runtime.
